firstpartydata/first_party_data_test.go

```diff
@@ -0,0 +1,552 @@
+				appContentDataKey:  nil,
+			},
+		},
+	}
+	for _, test := range testCases {
+
+		var req openrtb2.BidRequest
+		err := json.Unmarshal(test.input, &req)
+		assert.NoError(t, err, "Error should be nil")
+
+		res := ExtractOpenRtbGlobalFPD(&req)
+
+		resReq, err := json.Marshal(req)
+		assert.NoError(t, err, "Error should be nil")
+
+		assert.JSONEq(t, string(test.output), string(resReq), "Result request is incorrect")
+		assert.Equal(t, test.expectedFpdData[siteContentDataKey], res[siteContentDataKey], "siteContentData data is incorrect")
+		assert.Equal(t, test.expectedFpdData[userDataKey], res[userDataKey], "userData is incorrect")
+		assert.Equal(t, test.expectedFpdData[appContentDataKey], res[appContentDataKey], "appContentData is incorrect")
+
+	}
+}
+
+func TestPreprocessFPD(t *testing.T) {
+
+	if specFiles, err := ioutil.ReadDir("./tests/preprocessfpd"); err == nil {
+		for _, specFile := range specFiles {
+			fileName := "./tests/preprocessfpd/" + specFile.Name()
+
+			fpdFile, err := loadFpdFile(fileName)
+			if err != nil {
+				t.Errorf("Unable to load file: %s", fileName)
+			}
+			var extReq openrtb_ext.ExtRequestPrebid
+			err = json.Unmarshal(fpdFile.InputRequestData, &extReq)
+			if err != nil {
+				t.Errorf("Unable to unmarshal input request: %s", fileName)
+			}
+			fpdData, reqExtPrebid := ExtractBidderConfigFPD(extReq)
+
+			assert.Nil(t, reqExtPrebid.BidderConfigs, "Bidder specific FPD config should be removed from request")
+
+			assert.Equal(t, len(fpdFile.BiddersFPD), len(fpdData), "Incorrect fpd data")
+
+			for k, v := range fpdFile.BiddersFPD {
+
+				if v.Site != nil {
+					tempSiteExt := fpdData[k].Site.Ext
+					jsonutil.DiffJson(t, "site.ext is incorrect", v.Site.Ext, tempSiteExt)
+					//compare extensions first and the site objects without extensions
+					//in case two or more bidders share same config(pointer), ext should be returned back
+					v.Site.Ext = nil
+					fpdData[k].Site.Ext = nil
+					assert.Equal(t, v.Site, fpdData[k].Site, "Incorrect site fpd data")
+					fpdData[k].Site.Ext = tempSiteExt
+				}
+
+				if v.App != nil {
+
+					tempAppExt := fpdData[k].App.Ext
+					jsonutil.DiffJson(t, "app.ext is incorrect", v.App.Ext, tempAppExt)
+					//compare extensions first and the app objects without extensions
+					v.App.Ext = nil
+					fpdData[k].App.Ext = nil
+					assert.Equal(t, v.App, fpdData[k].App, "Incorrect app fpd data")
+					fpdData[k].App.Ext = tempAppExt
+				}
+
+				if v.User != nil {
+					tempUserExt := fpdData[k].User.Ext
+					jsonutil.DiffJson(t, "user.ext is incorrect", v.User.Ext, tempUserExt)
+					//compare extensions first and the user objects without extensions
+					v.User.Ext = nil
+					fpdData[k].User.Ext = nil
+					assert.Equal(t, v.User, fpdData[k].User, "Incorrect user fpd data")
+					fpdData[k].User.Ext = tempUserExt
+				}
+
+			}
+		}
+	}
+}
+
+func TestBuildResolvedFPDForBidders(t *testing.T) {
```

Comment:
Please rename this function to `TestResolveFPDData`

Reply:
Renamed